src/serializer/Emitter.js

```diff
@@ -343,6 +358,11 @@ export class Emitter {
         result = recurse(val.$Description);
         if (result !== undefined) return result;
       }
+    } else if (val instanceof ArrayValue && ArrayValue.isIntrinsicAndHasWidenedNumericProperty(val)) {
```

Comment:
I think the recursing on `val.$Prototype` in the `ObjectValue` case should also apply to `ArrayValue`. Maybe we are missing a test case that redefines an array's prototype?

Reply:
(and has a widened numeric property)

Reply:
We don't do that currently for arrays as the kind is `Array` which skips the check. Otherwise I would have kept that logic in too.